src/shogun/base/SGObject.h

```diff
@@ -1112,6 +1138,12 @@ CSGObject* get_dispatch_all_base_types(const CSGObject* obj, const std::string&
 		return result;
 	if (auto* result = get_if_possible<CFeatures>(obj, name, how))
 		return result;
+	if (auto* result = get_if_possible<CMachine>(obj, name, how))
+		return result;
+	if (auto* result = get_if_possible<CLabels>(obj, name, how))
```

Comment:
Rather than extending this list, we eventually want to have a single line call here that utilises a global list of base classes. But maybe we can do that another in another PR
@gf712 knows what to do :)